content/docs/1.10.0/advanced-resources/rebuilding/_index.md

```diff
@@ -1,4 +1,186 @@
+
+Rebuilding will not start when:
+
+- The volume is being migrating to another node.
+- The volume is old restore/DR volumes.
+- The volume is expanding size.
+
+## Replica Rebuilding Workflow
+
+Replica rebuilding may be triggered under the following scenarios:
+
+- Node reboot or eviction.
+- A replica becomes unhealthy or deleted.
+
+ {{< figure alt="Replica Rebuilding Flow Diagram" src="/img/diagrams/architecture/replica-rebuilding-flow.png" >}}
+
+1. Mark the target replica with `WO` (Write-Only) mode.
+2. System will create a new snapshot as a reference point of the volume-head for data integrity check.
+3. Get a synchronizing file list of the volume head and snapshot files.
+4. Launch a receiver server in the target replica for each snapshot, then ask the source replica to synchronize data.
+5. For each snapshot synchronization,
+    - Check whether the snapshot file exists in the target replica data directory
+        - If **NO**, directly transfer all data from the source replica to the destination replica. [Full Replica Rebuilding](#full-replica-rebuilding)
+        - If **YES**, check whether the value of snapshot checksum files between the target replica and the source replica exists and is identical:
+            - If **YES**, Longhorn skips the corresponding snapshot data transfer, which reduces the CPU usage, disk IO, network IO, and rebuilding time. [Fast Replica Rebuilding](#fast-replica-rebuilding)
+            - If **NO**, Longhorn calculates and compares block-by-block checksums for the snapshot file between the target replica and the source replica (The checksum is computed using SHA-512 method). If there is a mismatch, the corresponding data block will be synced. [Delta Replica Rebuilding](#delta-replica-rebuilding)
+
+### Full Replica Rebuilding
+
+If the replica is unrecoverable or replenished, Longhorn will synchronize the whole data from the healthy replica. Reconstruct the replica by sending the full snapshot chain.
+
+Full replica rebuilding will waste network bandwidth and generate numerous disk WRITE operations on the target replica node, but it is the only way to rebuild a replica if there is no data on the target node.
+
+### Delta Replica Rebuilding
+
+- This is available for failed replica reusage only, and there is an existing snapshot file (with the same name) in the failed replica data directory
+- There is no checksum/checksum file for the snapshot file. It's like a fallback of fast rebuilding
+
+- **Pros**:
+  - Reduce network bandwidth consumption.
+- **Cons**:
+  - Increased CPU overhead.
+  - Rebuilding time is influenced by CPU performance.
+
+### Fast Replica Rebuilding
+
+Fast rebuilding is enabled when:
+
+- The fast replica building setting is enabled:
+  - `fast-replica-rebuild-enabled: true`
+- Snapshot checksum files are created:
+  - `snapshot-data-integrity`: `enabled`, and a scheduled job will calculate checksums of all snapshots in a configured interval or
```

Comment:
```suggestion
  - `snapshot-data-integrity` is set to `enabled`: a scheduled job will calculate checksums of all snapshots in a configured interval or
```

> a configured interval 

Mention which setting determines the interval

Reply:
Modified. Thanks.